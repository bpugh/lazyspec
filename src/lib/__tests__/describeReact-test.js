const fs = require('fs');

const describeReact = require('../describeReact');

function expectDescription(moduleName) {
  const unitPath = `spec/fixtures/${moduleName}.js`;
  const src = fs.readFileSync(unitPath).toString();
  const description = describeReact(unitPath, src);

  expect(description).toMatchSnapshot();
}

describe('describeReact', () => {
  it('should describe simple exports', () => {
    expect(
      () => expectDescription('esExports'),
    ).toThrow();
  });

  it('should describe an ES5 component', () => {
    expectDescription('CreateClass');
  });

  it('should describe an ES6 component', () => {
    expectDescription('SafeComponent');
  });

  it('should describe an unsafe ES6 component', () => {
    expectDescription('UnsafeComponent');
  });

  it('should describe an ES6 with Flow component', () => {
    expectDescription('FlowComponent');
  });

  it('should describe a stateless function component', () => {
    expectDescription('StatelessFunction');
  });

  // Not currently supported by react-docgen
  it('should not describe multi-exports', () => {
    expect(
      () => expectDescription('MultipleComponents'),
    ).toThrowError('Multiple exported component definitions found.');
  });
});
