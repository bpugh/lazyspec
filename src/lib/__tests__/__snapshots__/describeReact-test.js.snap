exports[`describeReact should describe a stateless function component 1`] = `
Object {
  "description": "",
  "methods": Array [],
}
`;

exports[`describeReact should describe an ES5 component 1`] = `
Object {
  "description": "",
  "methods": Array [
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "_setup",
      "params": Array [],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "show",
      "params": Array [],
      "returns": null,
    },
  ],
  "props": Object {
    "disabled": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "height": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "label": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "level": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "\'primary\'",
          },
          Object {
            "computed": false,
            "value": "\'secondary\'",
          },
          Object {
            "computed": false,
            "value": "\'tertiary\'",
          },
          Object {
            "computed": false,
            "value": "\'quaternary\'",
          },
          Object {
            "computed": false,
            "value": "\'text\'",
          },
          Object {
            "computed": false,
            "value": "\'icon\'",
          },
        ],
      },
    },
  },
}
`;

exports[`describeReact should describe an ES6 component 1`] = `
Object {
  "description": "",
  "methods": Array [],
  "props": Object {
    "children": Object {
      "description": "",
      "required": true,
      "type": Object {
        "name": "element",
      },
    },
    "disabled": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "height": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "label": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "level": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "\'primary\'",
          },
          Object {
            "computed": false,
            "value": "\'secondary\'",
          },
          Object {
            "computed": false,
            "value": "\'tertiary\'",
          },
          Object {
            "computed": false,
            "value": "\'quaternary\'",
          },
          Object {
            "computed": false,
            "value": "\'text\'",
          },
          Object {
            "computed": false,
            "value": "\'icon\'",
          },
        ],
      },
    },
  },
}
`;

exports[`describeReact should describe an ES6 with Flow component 1`] = `
Object {
  "description": "",
  "methods": Array [],
  "props": Object {
    "children": Object {
      "description": "",
      "flowType": Object {
        "name": "any",
      },
      "required": true,
    },
    "disabled": Object {
      "description": "",
      "flowType": Object {
        "name": "boolean",
      },
      "required": false,
    },
    "label": Object {
      "description": "",
      "flowType": Object {
        "name": "string",
      },
      "required": true,
    },
  },
}
`;

exports[`describeReact should describe an unsafe ES6 component 1`] = `
Object {
  "description": "",
  "methods": Array [],
  "props": Object {
    "foo": Object {
      "description": "",
      "required": true,
      "type": Object {
        "name": "instanceOf",
        "value": "React",
      },
    },
  },
}
`;
